flaskbb/cli/plugins.py

```diff
@@ -33,98 +21,30 @@ def plugins():
-                fg="green", bold=True)
-
-
-@plugins.command("install")
-@click.argument("plugin_identifier")
-def install_plugin(plugin_identifier):
-    """Installs a new plugin."""
-    validate_plugin(plugin_identifier)
-    plugin = get_plugin_from_all(plugin_identifier)
-    click.secho("[+] Installing plugin {}...".format(plugin.name), fg="cyan")
-    try:
-        plugin_manager.install_plugins([plugin])
-    except Exception as e:
-        click.secho("[-] Couldn't install plugin because of following "
-                    "exception: \n{}".format(e), fg="red")
-
-
-@plugins.command("uninstall")
-@click.argument("plugin_identifier")
-def uninstall_plugin(plugin_identifier):
-    """Uninstalls a plugin from FlaskBB."""
-    validate_plugin(plugin_identifier)
-    plugin = get_plugin_from_all(plugin_identifier)
-    click.secho("[+] Uninstalling plugin {}...".format(plugin.name), fg="cyan")
-    try:
-        plugin_manager.uninstall_plugins([plugin])
-    except AttributeError:
-        pass
-
-
-@plugins.command("remove")
-@click.argument("plugin_identifier")
-@click.option("--force", "-f", default=False, is_flag=True,
-              help="Removes the plugin without asking for confirmation.")
-def remove_plugin(plugin_identifier, force):
-    """Removes a plugin from the filesystem."""
-    validate_plugin(plugin_identifier)
-    if not force and not \
-            click.confirm(click.style("Are you sure?", fg="magenta")):
-        sys.exit(0)
-
-    plugin = get_plugin_from_all(plugin_identifier)
-    click.secho("[+] Uninstalling plugin {}...".format(plugin.name), fg="cyan")
-    try:
-        plugin_manager.uninstall_plugins([plugin])
-    except Exception as e:
-        click.secho("[-] Couldn't uninstall plugin because of following "
-                    "exception: \n{}".format(e), fg="red")
-        if not click.confirm(click.style(
-            "Do you want to continue anyway?", fg="magenta")
-        ):
-            sys.exit(0)
-
-    click.secho("[+] Removing plugin from filesystem...", fg="cyan")
-    shutil.rmtree(plugin.path, ignore_errors=False, onerror=None)
-
-
 @plugins.command("list")
 def list_plugins():
     """Lists all installed plugins."""
     click.secho("[+] Listing all installed plugins...", fg="cyan")
 
-    # This is subject to change as I am not happy with the current
-    # plugin system
-    enabled_plugins = get_enabled_plugins()
-    disabled_plugins = set(get_all_plugins()) - set(enabled_plugins)
+    enabled_plugins = current_app.pluggy.list_plugin_distinfo()
     if len(enabled_plugins) > 0:
         click.secho("[+] Enabled Plugins:", fg="blue", bold=True)
         for plugin in enabled_plugins:
+            # plugin[0] is the module
+            plugin = plugin[1]
             click.secho("    - {} (version {})".format(
-                plugin.name, plugin.version), bold=True
+                plugin.key, plugin.version), bold=True
             )
+
+    # TODO: is there a better way for doing this?
```

Comment:
Yes, wrap this using `with_appcontext` and load the information off of `current_app.pluggy`, with a fix I pushed this'll end up loading plugins twice which *might* cause issues.

Reply:
Actually, that shouldn't be needed anymore as we keep track of disabled plugins :) ``pluggy.list_disabled_plugins()`` should provide all information needed for this.